src/ui/map.js

```diff
@@ -2933,6 +2935,10 @@ class Map extends Camera {
             // all tiles held for fading. If we didn't do this, the tiles
             // would just sit in the SourceCaches until the next render
             this.style._releaseSymbolFadeTiles();
+
+            // `_isInitialGeoJSONData` is used to temporarily set `fadeDuration` to zero
+            // as a workaround to prevent placement from fading symbols when triggered by `setData`.
+            if (this._isInitialGeoJSONData && this.style.loaded()) this._isInitialGeoJSONData = false;
```

Comment:
Is `this.style.loaded()` the right check here? Have you seen a case where `this.style.loaded()` is true, but the tile data isn't yet represented and shown on the tiles?